### PR TITLE
Tables expanding

### PR DESCRIPTION
**Pull request by richmccartney**
## Done

- Creates an expanding table cell that can be used to hide/show content

## QA

- Pull branch and run `./run`
- Check code and view the new pattern at http://0.0.0.0:8102/vanilla-dashboard-theme/examples/patterns/table-expanding/table-expanding/

### CODE REVIEW

**Comment by anthonydillon** [comment]
There are some linting issues here. Could you resolve them first before reviewing.

**Comment by richmccartney** [comment]
@anthonydillon Ah thanks, pushed if you want to review it now 😄

File: examples/patterns/table-expanding/table-expanding.html [hunk 1]
```diff
@@ -0,0 +1,62 @@
+---
+layout: default
+title: Table expanding
+category: _patterns
+---
+
+<table class="p-table-expanding" role="grid">
+    <thead>
+        <tr role="row">
+            <th scope="col" role="columnheader" id="t-name" aria-sort="none">Name</th>
```
Comment (anthonydillon) [inline examples/patterns/table-expanding/table-expanding.html]: Please remove the `role="columnheader"` here as it is unnecessary on `th`.

File: examples/patterns/table-expanding/table-expanding.html [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+            <th scope="col" role="columnheader" id="t-users" aria-sort="none" class="u-align--right">Users</th>
+            <th scope="col" role="columnheader" id="t-units" aria-sort="none" class="u-align--right">Units</th>
+            <th scope="col" role="columnheader" id="t-revenue" aria-sort="none" class="u-align--right">Actions</th>
+        </tr>
+    </thead>
+    <tbody>
+        <tr role="row">
+            <td role="rowheader" aria-label="Name">Grape</td>
+            <td role="gridcell" aria-label="Users"class="u-align--right">8</td>
```
Comment (anthonydillon) [inline examples/patterns/table-expanding/table-expanding.html]: Add space `aria-label="Users"class="u-align--right"`

File: examples/patterns/table-expanding/table-expanding.html [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            <td role="rowheader" aria-label="Name">Melon</td>
+            <td role="gridcell" aria-label="Users" class="u-align--right">12</td>
+            <td role="gridcell" aria-label="Units" class="u-align--right">23</td>
+            <td></td>
+        </tr>
+        <tr role="row">
+            <td role="rowheader" aria-label="Name">Apple</td>
+            <td role="gridcell" aria-label="Users" class="u-align--right">9</td>
+            <td role="gridcell" aria-label="Units" class="u-align--right">17</td>
+            <td role="gridcell"class="u-align--right">
```
Comment (anthonydillon) [inline examples/patterns/table-expanding/table-expanding.html]: Add space before class.

File: examples/patterns/table-expanding/table-expanding.html [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+<script>
+function toggleHidden(id) {
+  var attr = document.getElementById(id).attributes;
+
+  if (attr['aria-hidden'].value == "true") {
+    document.getElementById(id).setAttribute("aria-hidden", "false");
+  } else {
+    document.getElementById(id).setAttribute("aria-hidden", "true");
+  }
```
Comment (anthonydillon) [inline examples/patterns/table-expanding/table-expanding.html]: Couldn't this functionality be added as a single line onClick function on the button?

**Comment by richmccartney** [comment]
@anthonydillon can you take a second look now please.

**Pull request merged**